cpp/src/arrow/dataset/scanner.h

```diff
@@ -384,6 +381,23 @@ class ARROW_DS_EXPORT ScannerBuilder {
   /// This option provides a control limiting the memory owned by any RecordBatch.
   Status BatchSize(int64_t batch_size);
 
+  /// \brief Set the number of batches to read aheadw ithin a file.
+  ///
+  /// \param[in] batch_readahead How many batches to read ahead within a file,
+  ///  might not work for all formats. refer to comments above.
```

Comment:
```suggestion
  ///  might not work for all formats.
```

These comments are used to generate C++ API docs.  I'm not sure what comments you are hoping to refer to but this phrase "refer to comments above" will not make sense in the generated docs I think.